frontend/src/core/editor/components/EditorMainAction.js

```diff
@@ -0,0 +1,107 @@
+/* @flow */
+
+import * as React from 'react';
+import { Localized } from '@fluent/react';
+
+import type { EntityTranslation } from 'core/api';
+import type { ChangeOperation, HistoryState } from 'modules/history';
+import type { Translation } from '..';
+
+
+type Props = {
+    activeTranslation: EntityTranslation,
+    initialTranslation: Translation,
+    isTranslator: boolean,
+    forceSuggestions: boolean,
+    history: HistoryState,
+    translation: Translation,
+    sendTranslation: () => void,
+    updateTranslationStatus: (number, ChangeOperation, ?boolean) => void,
+};
+
+
+/**
+ * Render the main action button of the Editor.
+ *
+ * This component renders a button to "Save", "Suggest" or "Approve" a translation.
+ *
+ * It renders "Approve" if the translation in the Editor is the same as an
+ * already existing translation for that same entity and locale, and the user
+ * has permission to approve.
+ * Otherwise, if the "force suggestion" user setting is on, it renders "Suggest".
+ * Otherwise, it renders "Save".
+ */
+export default function EditorMainAction(props: Props) {
+    const {
+        activeTranslation,
+        initialTranslation,
+        isTranslator,
+        forceSuggestions,
+        history,
+        translation,
+        sendTranslation,
+        updateTranslationStatus,
+    } = props;
+
+    let sameTranslationAs = null;
+    if (translation) {
+        if (translation === initialTranslation) {
+            sameTranslationAs = activeTranslation;
+        }
+        else if (history.translations.length) {
+            sameTranslationAs = history.translations.find(t => t.string === translation)
+        }
+    }
+
+    function approveTranslation() {
+        if (sameTranslationAs) {
```

Comment:
Is this check necessary? Seems like `approveTranslation()` can only be called if the condition is true.

Reply:
It's needed for Flow. :sweat_smile: